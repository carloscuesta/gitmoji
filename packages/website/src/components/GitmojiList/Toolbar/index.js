// @flow
import { type Element, useEffect, useRef } from 'react'

import ListModeSelector from './ListModeSelector'
import ThemeSelector from './ThemeSelector'
import styles from './styles.module.css'

type Props = {
  isListMode: boolean,
  searchInput: ?string,
  setIsListMode: Function,
  setSearchInput: Function,
}

const isMacOs = () => {
  return typeof window === 'undefined'
    ? true
    : window.navigator.platform.toUpperCase().indexOf('MAC') >= 0
}

const Toolbar = (props: Props): Element<'div'> => {
  const searchInputRef = useRef(null)

  useEffect(() => {
    const keyboardEventListener = (event: KeyboardEvent) => {
      const searchInput = searchInputRef.current
      if (
        searchInput &&
        (event.ctrlKey || event.metaKey) &&
        event.key === 'k'
      ) {
        event.preventDefault()
        searchInput.focus()
      }
    }

    if (typeof window !== 'undefined') {
      document.addEventListener('keydown', keyboardEventListener, false)
    }

    return () => {
      document.removeEventListener('keydown', keyboardEventListener, false)
    }
  }, [])

  return (
    <div className={styles.container}>
      <div className={styles.inputWrapper}>
        <input
          className={styles.searchInput}
          ref={searchInputRef}
          name="searchInput"
          onChange={(event) => props.setSearchInput(event.target.value)}
          placeholder="Search your gitmoji..."
          type="text"
          value={props.searchInput}
        />

        <kbd className={styles.kbd}>{isMacOs() ? 'âŒ˜' : 'Ctrl'} K</kbd>
      </div>

      <div className={styles.actionsContainer}>
        <ThemeSelector />

        <ListModeSelector
          isListMode={props.isListMode}
          setIsListMode={props.setIsListMode}
        />
      </div>
    </div>
  )
}

export default Toolbar
