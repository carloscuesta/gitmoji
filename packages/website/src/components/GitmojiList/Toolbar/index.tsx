import { useEffect, useRef } from 'react'
import dynamic from 'next/dynamic'

import ListModeSelector from './ListModeSelector'
import ThemeSelector from './ThemeSelector'
const Kbd = dynamic(() => import('./Kbd'), { ssr: false })
import styles from './styles.module.css'

type Props = {
  isListMode: boolean
  searchInput?: string
  setIsListMode: (searchInput: boolean) => void
  setSearchInput: (searchInput: string) => void
}

const Toolbar = (props: Props) => {
  const searchInputRef = useRef<HTMLInputElement>(null)

  useEffect(() => {
    const keyboardEventListener = (event: KeyboardEvent) => {
      const searchInput = searchInputRef.current
      if (
        searchInput &&
        (event.ctrlKey || event.metaKey) &&
        event.key === 'k'
      ) {
        event.preventDefault()
        searchInput.focus()
      }
    }

    if (typeof window !== 'undefined') {
      document.addEventListener('keydown', keyboardEventListener, false)
    }

    return () => {
      document.removeEventListener('keydown', keyboardEventListener, false)
    }
  }, [])

  return (
    <div className={styles.container}>
      <div className={styles.inputWrapper}>
        <input
          className={styles.searchInput}
          ref={searchInputRef}
          name="searchInput"
          onChange={(event) => props.setSearchInput(event.target.value)}
          placeholder="Search your gitmoji..."
          type="search"
          value={props.searchInput}
        />

        <Kbd />
      </div>

      <div className={styles.actionsContainer}>
        <ThemeSelector />

        <ListModeSelector
          isListMode={props.isListMode}
          setIsListMode={props.setIsListMode}
        />
      </div>
    </div>
  )
}

export default Toolbar
