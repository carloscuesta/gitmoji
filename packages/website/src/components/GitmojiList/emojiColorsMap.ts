export default {
  'adhesive-bandage': '#fbcfb7',
  alembic: '#7f39fb',
  alien: '#c5e763',
  ambulance: '#fb584a',
  dizzy: '#ffdb3a',
  'arrow-down': '#ef5350',
  'arrow-up': '#00e676',
  art: '#ff7281',
  beers: '#fbb64b',
  bento: '#ff5864',
  bookmark: '#80deea',
  boom: '#f94f28',
  bug: '#8cd842',
  'building-construction': '#ffe55f',
  bulb: '#ffce49',
  'busts-in-silhouette': '#ffce49',
  'camera-flash': '#00a9f0',
  'card-file-box': '#c5e763',
  'chart-with-upwards-trend': '#cedae6',
  'children-crossing': '#ffce49',
  'clown-face': '#ff7281',
  'construction-worker': '#64b5f6',
  construction: '#ffb74d',
  egg: '#77e856',
  fire: '#ff9d44',
  'globe-with-meridians': '#e7f4ff',
  'goal-net': '#c7cb12',
  'green-heart': '#c5e763',
  hammer: '#ffc400',
  coffin: '#d9e3e8',
  'heavy-minus-sign': '#ef5350',
  'heavy-plus-sign': '#00e676',
  iphone: '#40c4ff',
  label: '#cb63e6',
  lipstick: '#80deea',
  lock: '#ffce49',
  'closed-lock-with-key': '#83beec',
  'loud-sound': '#23b4d2',
  mag: '#ffe55f',
  memo: '#00e676',
  mute: '#e6ebef',
  'ok-hand': '#c5e763',
  package: '#fdd0ae',
  'page-facing-up': '#d9e3e8',
  'passport-control': '#4dc6dc',
  pencil: '#ffce49',
  pencil2: '#ffce49',
  poop: '#a78674',
  pushpin: '#39c2f1',
  recycle: '#77e856',
  rewind: '#56d1d8',
  rocket: '#00a9f0',
  'rotating-light': '#536dfe',
  'safety-vest': '#f2ad52',
  'see-no-evil': '#8bdfe7',
  seedling: '#c5e763',
  sparkles: '#ffe55f',
  'speech-balloon': '#cedae6',
  stethoscope: '#77e856',
  tada: '#f74d5f',
  'test-tube': '#fb584a',
  'triangular-flag-on-post': '#ffce49',
  truck: '#ef584a',
  'twisted-rightwards-arrows': '#56d1d8',
  wastebasket: '#d9e3e8',
  wheelchair: '#00b1fb',
  'white-check-mark': '#77e856',
  wrench: '#ffc400',
  zap: '#40c4ff',
  'monocle-face': '#ffe55f',
  necktie: '#83beec',
  bricks: '#ff6723',
  technologist: '#86B837',
  'money-with-wings': '#b3c0b1',
  thread: '#ffbe7b',
} as const
