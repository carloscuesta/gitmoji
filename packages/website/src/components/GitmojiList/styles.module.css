.gitmojiCode {
  padding: 0 4px;
  border-radius: 4px;
  background-color: var(--notificationEmojiCodeColor);
  color: var(--notificationText);
}

.closeButton {
  width: 20px;
  height: 20px;
  position: absolute;
  right: 4px;
  top: 4px;
  overflow: hidden;
  text-indent: 100%;
  cursor: pointer;
  backface-visibility: hidden;
}

.closeButton:hover,
.closeButton:focus {
  outline: none;
}

.closeButton::before,
.closeButton::after {
  content: '';
  position: absolute;
  width: 3px;
  height: 60%;
  top: 50%;
  left: 50%;
  background: var(--notificationText);
}

.closeButton::before {
  transform: translate(-50%, -50%) rotate(45deg);
}

.closeButton::after {
  transform: translate(-50%, -50%) rotate(-45deg);
}
