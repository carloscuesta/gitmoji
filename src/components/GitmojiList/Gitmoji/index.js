import React from 'react'

type Props = {
  code: string,
  description: string,
  emoji: string,
  name: string,
}

const Gitmoji = (props: Props) => (
  <article className="emoji col-xs-12 col-sm-6 col-md-3">
    <div className="emoji-card">
      <header className={`${props.name} emoji-header`}>
        <span
          className="emoji-icon gitmoji-emoji"
          data-clipboard-text={props.emoji}
        >
          {props.emoji}
        </span>
      </header>
      <div className="emoji-info">
        <div className="gitmoji-code" data-clipboard-text={props.code}>
          <code>{props.code}</code>
        </div>
        <p>{props.description}</p>
      </div>
    </div>
  </article>
)

export default Gitmoji
