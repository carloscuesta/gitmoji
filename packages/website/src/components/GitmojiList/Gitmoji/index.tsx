import emojiColorsMap from '../emojiColorsMap'
import styles from './styles.module.css'

type Props = {
  code: string
  description: string
  emoji: string
  isListMode: boolean
  name: keyof typeof emojiColorsMap
}

const Gitmoji = (props: Props) => {
  const style = {
    '--emojiColor': emojiColorsMap[props.name],
  } as React.CSSProperties

  return (
    <article
      style={style}
      className={`${styles.emoji} col-xs-12 col-sm-6 ${
        props.isListMode ? 'col-md-4' : 'col-md-3'
      }`}
    >
      <div
        className={`${styles.card} ${props.isListMode ? styles.cardList : ''}`}
      >
        <header className={`${styles.cardHeader}`}>
          <button
            type="button"
            className={`gitmoji-clipboard-emoji ${styles.gitmoji}`}
            data-clipboard-text={props.emoji}
          >
            {props.emoji}
          </button>
        </header>
        <div className={styles.gitmojiInfo}>
          <button
            className={`gitmoji-clipboard-code ${styles.gitmojiCode}`}
            data-clipboard-text={props.code}
            tabIndex={-1}
            type="button"
          >
            <code>
              {replaceWithJSX(
                props.code,
                '_',
                <>
                  _<wbr />
                </>,
              )}
            </code>
          </button>
          <p>{props.description}</p>
        </div>
      </div>
    </article>
  )
}

const replaceWithJSX = (text: string, find: string, replace: JSX.Element) => {
  const nodes: (string | JSX.Element)[] = text.split(find)
  const first = nodes.shift()

  return nodes
    .reduce((newNodes, part) => [...newNodes, replace, part], [first])
    .map((el, index) => <span key={index}>{el}</span>)
}

export default Gitmoji
