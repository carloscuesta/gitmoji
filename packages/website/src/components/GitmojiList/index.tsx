import { useEffect, useState } from 'react'
import Clipboard from 'clipboard'
import { useRouter } from 'next/router'
import type { Gitmoji as GitmojiType } from 'gitmojis'
import toast from 'react-hot-toast'

import Gitmoji from './Gitmoji'
import Toolbar from './Toolbar'
import useLocalStorage from './hooks/useLocalStorage'
import styles from './styles.module.css'

type Props = {
  gitmojis: readonly GitmojiType[]
}

const GitmojiList = (props: Props) => {
  const router = useRouter()
  const [searchInput, setSearchInput] = useState('')
  const [isListMode, setIsListMode] = useLocalStorage('isListMode', false)

  const gitmojis = searchInput
    ? props.gitmojis.filter(({ emoji, code, description }) => {
        const lowerCasedSearch = searchInput.toLowerCase()

        return (
          code.includes(lowerCasedSearch) ||
          description.toLowerCase().includes(lowerCasedSearch) ||
          emoji == searchInput
        )
      })
    : props.gitmojis

  useEffect(() => {
    if (router.query.search) {
      setSearchInput(router.query.search as string)
    }
  }, [router.query.search])

  useEffect(() => {
    if (router.query.search && !searchInput) {
      router.push('/', undefined, { shallow: true })
    }
  }, [searchInput])

  useEffect(() => {
    const clipboard = new Clipboard(
      '.gitmoji-clipboard-emoji, .gitmoji-clipboard-code',
    )

    clipboard.on('success', function (e) {
      toast(
        (t) => (
          <span className={styles.notification}>
            <p>
              Hey! Gitmoji <span className={styles.gitmojiCode}>{e.text}</span>{' '}
              copied to the clipboard ðŸ˜œ
            </p>
            <span
              className={styles.closeButton}
              onClick={() => toast.dismiss(t.id)}
            />
          </span>
        ),
        {
          id: 'clipboard',
          style: {
            background: '#ff5a79',
            color: '#ffffff',
            fontWeight: 600,
            fontSize: '90%',
          },
        },
      )
    })

    return () => clipboard.destroy()
  }, [])

  return (
    <div className="row" id="gitmoji-list">
      <div className="col-xs-12">
        <Toolbar
          isListMode={isListMode}
          searchInput={searchInput}
          setIsListMode={setIsListMode}
          setSearchInput={setSearchInput}
        />
      </div>

      {gitmojis.length === 0 ? (
        <h2>No gitmojis found for search: {searchInput}</h2>
      ) : (
        gitmojis.map((gitmoji, index) => (
          <Gitmoji
            code={gitmoji.code}
            description={gitmoji.description}
            emoji={gitmoji.emoji}
            isListMode={isListMode}
            key={index}
            // @ts-expect-error: This should be replaced with something like:
            // typeof gitmojis[number]['name'] but JSON can't be exported `as const`
            name={gitmoji.name}
          />
        ))
      )}
    </div>
  )
}

export default GitmojiList
