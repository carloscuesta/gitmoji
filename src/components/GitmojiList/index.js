// @flow
import React, { type Element } from 'react'
import Clipboard from 'clipboard'

import Gitmoji from './Gitmoji'

type Props = {
  gitmojis: Array<{
    code: string,
    description: string,
    emoji: string,
    name: string,
  }>,
}

const GitmojiList = (props: Props): Element<'div'> => {
  React.useEffect(() => {
    const clipboard = new Clipboard('.gitmoji-code, .gitmoji-emoji')

    clipboard.on('success', function (e) {
      window.ga('send', 'event', 'Gitmoji', 'Copy to Clipboard')

      const notification = new window.NotificationFx({
        message: e.trigger.classList.contains('gitmoji-emoji')
          ? `<p>Hey! Gitmoji emoji ${e.text} copied to the clipboard ðŸ˜œ</p>`
          : `<p>Hey! Gitmoji <span class="gitmoji-code">${e.text}</span> copied to the clipboard ðŸ˜œ</p>`,
        layout: 'growl',
        effect: 'scale',
        type: 'notice',
        ttl: 2000,
      })

      notification.show()
    })

    return () => clipboard.destroy()
  }, [])

  return (
    <div className="row center-xs" id="gitmoji-list">
      {props.gitmojis.map((gitmoji, index) => (
        <Gitmoji
          code={gitmoji.code}
          description={gitmoji.description}
          emoji={gitmoji.emoji}
          key={index}
          name={gitmoji.name}
        />
      ))}
    </div>
  )
}

export default GitmojiList
