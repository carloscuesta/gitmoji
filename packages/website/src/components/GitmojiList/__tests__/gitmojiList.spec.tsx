import { useRouter } from 'next/router'
import renderer from 'react-test-renderer'

import GitmojiList from '../index'
import * as stubs from './stubs'

jest.mock('next/router', () => ({
  query: {},
  useRouter: jest.fn().mockImplementation(() => ({
    query: {},
    push: jest.fn(),
  })),
}))

const useRouterMock = useRouter as jest.Mock

describe('GitmojiList', () => {
  describe('when is not list mode', () => {
    it('should render the component', () => {
      const wrapper = renderer.create(<GitmojiList {...stubs.props} />)
      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when is list mode', () => {
    it('should render the component', () => {
      const wrapper = renderer.create(<GitmojiList {...stubs.props} />)
      const instance = wrapper.root

      renderer.act(() => {
        instance.findAllByType('button')[1].props.onClick()
      })

      expect(wrapper).toMatchSnapshot()
    })
  })

  describe('when user search the fire gitmoji', () => {
    beforeAll(() => {
      useRouterMock.mockReturnValue(stubs.routerMock())
    })

    it('should find the fire gitmoji by code', () => {
      const wrapper = renderer.create(<GitmojiList {...stubs.props} />)
      const instance = wrapper.root
      const query = 'Fire'

      renderer.act(() => {
        instance
          .findByType('input')
          .props.onChange({ target: { value: query } })
      })

      expect(instance.findAllByType('article').length).toEqual(1)
    })

    it('should find the fire gitmoji by description', () => {
      const wrapper = renderer.create(<GitmojiList {...stubs.props} />)
      const instance = wrapper.root
      const query = 'remove'

      renderer.act(() => {
        instance
          .findByType('input')
          .props.onChange({ target: { value: query } })
      })

      expect(instance.findAllByType('article').length).toEqual(1)
    })

    it('should find the fire gitmoji by emoji', () => {
      const wrapper = renderer.create(<GitmojiList {...stubs.props} />)
      const instance = wrapper.root
      const query = 'ğŸ”¥'

      renderer.act(() => {
        instance
          .findByType('input')
          .props.onChange({ target: { value: query } })
      })

      expect(instance.findAllByType('article').length).toEqual(1)
    })
  })

  describe('when search is provided by query string', () => {
    beforeAll(() => {
      useRouterMock.mockReturnValue(stubs.routerMock({ search: 'fire' }))
    })

    it('should set the search input value to query.search', () => {
      const wrapper = renderer.create(<GitmojiList {...stubs.props} />)
      const query = 'fire'

      renderer.act(() => {
        wrapper.update(<GitmojiList {...stubs.props} />)
      })

      expect(wrapper.root.findByType('input').props.value).toEqual(query)
    })

    describe('when the user deletes the search input', () => {
      it('should clear the query string', () => {
        const wrapper = renderer.create(<GitmojiList {...stubs.props} />)
        const instance = wrapper.root

        renderer.act(() => {
          instance.findByType('input').props.onChange({ target: { value: '' } })
        })

        expect(useRouterMock().push).toHaveBeenCalledWith('/', undefined, {
          shallow: true,
        })
      })
    })
  })
})
