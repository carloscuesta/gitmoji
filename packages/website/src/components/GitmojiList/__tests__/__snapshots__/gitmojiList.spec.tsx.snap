// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GitmojiList when is list mode should render the component 1`] = `
<div
  className="row"
  id="gitmoji-list"
>
  <div
    className="col-xs-12"
  >
    <div
      className="container"
    >
      <div
        className="inputWrapper"
      >
        <input
          className="searchInput"
          name="searchInput"
          onChange={[Function]}
          placeholder="Search your gitmoji..."
          type="search"
          value=""
        />
        <kbd
          className="kbd"
        >
          Ctrl
           K
        </kbd>
      </div>
      <div
        className="actionsContainer"
      >
        <button
          className="button"
          onClick={[Function]}
        >
          <svg
            className="icon icon-light"
          >
            <use
              xlinkHref="#icon-light"
            />
          </svg>
        </button>
        <div
          className="container"
        >
          <button
            className="button buttonActive"
            disabled={true}
            onClick={[Function]}
          >
            <svg
              className="icon icon-grid"
            >
              <use
                xlinkHref="#icon-grid"
              />
            </svg>
          </button>
          <button
            className="button "
            disabled={false}
            onClick={[Function]}
          >
            <svg
              className="icon icon-list"
            >
              <use
                xlinkHref="#icon-list"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <article
    className="emoji col-xs-12 col-sm-6 col-md-3"
    style={
      {
        "--emojiColor": "#ff7281",
      }
    }
  >
    <div
      className="card "
    >
      <header
        className="cardHeader"
      >
        <button
          className="gitmoji-clipboard-emoji gitmoji"
          data-clipboard-text="üé®"
          type="button"
        >
          üé®
        </button>
      </header>
      <div
        className="gitmojiInfo"
      >
        <button
          className="gitmoji-clipboard-code gitmojiCode"
          data-clipboard-text=":art:"
          tabIndex={-1}
          type="button"
        >
          <code>
            <span>
              :art:
            </span>
          </code>
        </button>
        <p>
          Improve structure / format of the code.
        </p>
      </div>
    </div>
  </article>
  <article
    className="emoji col-xs-12 col-sm-6 col-md-3"
    style={
      {
        "--emojiColor": "#40c4ff",
      }
    }
  >
    <div
      className="card "
    >
      <header
        className="cardHeader"
      >
        <button
          className="gitmoji-clipboard-emoji gitmoji"
          data-clipboard-text="‚ö°Ô∏è"
          type="button"
        >
          ‚ö°Ô∏è
        </button>
      </header>
      <div
        className="gitmojiInfo"
      >
        <button
          className="gitmoji-clipboard-code gitmojiCode"
          data-clipboard-text=":zap:"
          tabIndex={-1}
          type="button"
        >
          <code>
            <span>
              :zap:
            </span>
          </code>
        </button>
        <p>
          Improve performance.
        </p>
      </div>
    </div>
  </article>
  <article
    className="emoji col-xs-12 col-sm-6 col-md-3"
    style={
      {
        "--emojiColor": "#ff9d44",
      }
    }
  >
    <div
      className="card "
    >
      <header
        className="cardHeader"
      >
        <button
          className="gitmoji-clipboard-emoji gitmoji"
          data-clipboard-text="üî•"
          type="button"
        >
          üî•
        </button>
      </header>
      <div
        className="gitmojiInfo"
      >
        <button
          className="gitmoji-clipboard-code gitmojiCode"
          data-clipboard-text=":fire:"
          tabIndex={-1}
          type="button"
        >
          <code>
            <span>
              :fire:
            </span>
          </code>
        </button>
        <p>
          Remove code or files.
        </p>
      </div>
    </div>
  </article>
  <article
    className="emoji col-xs-12 col-sm-6 col-md-3"
    style={
      {
        "--emojiColor": "#8cd842",
      }
    }
  >
    <div
      className="card "
    >
      <header
        className="cardHeader"
      >
        <button
          className="gitmoji-clipboard-emoji gitmoji"
          data-clipboard-text="üêõ"
          type="button"
        >
          üêõ
        </button>
      </header>
      <div
        className="gitmojiInfo"
      >
        <button
          className="gitmoji-clipboard-code gitmojiCode"
          data-clipboard-text=":bug:"
          tabIndex={-1}
          type="button"
        >
          <code>
            <span>
              :bug:
            </span>
          </code>
        </button>
        <p>
          Fix a bug.
        </p>
      </div>
    </div>
  </article>
  <article
    className="emoji col-xs-12 col-sm-6 col-md-3"
    style={
      {
        "--emojiColor": "#fb584a",
      }
    }
  >
    <div
      className="card "
    >
      <header
        className="cardHeader"
      >
        <button
          className="gitmoji-clipboard-emoji gitmoji"
          data-clipboard-text="üöëÔ∏è"
          type="button"
        >
          üöëÔ∏è
        </button>
      </header>
      <div
        className="gitmojiInfo"
      >
        <button
          className="gitmoji-clipboard-code gitmojiCode"
          data-clipboard-text=":ambulance:"
          tabIndex={-1}
          type="button"
        >
          <code>
            <span>
              :ambulance:
            </span>
          </code>
        </button>
        <p>
          Critical hotfix.
        </p>
      </div>
    </div>
  </article>
  <article
    className="emoji col-xs-12 col-sm-6 col-md-3"
    style={
      {
        "--emojiColor": "#ffe55f",
      }
    }
  >
    <div
      className="card "
    >
      <header
        className="cardHeader"
      >
        <button
          className="gitmoji-clipboard-emoji gitmoji"
          data-clipboard-text="‚ú®"
          type="button"
        >
          ‚ú®
        </button>
      </header>
      <div
        className="gitmojiInfo"
      >
        <button
          className="gitmoji-clipboard-code gitmojiCode"
          data-clipboard-text=":sparkles:"
          tabIndex={-1}
          type="button"
        >
          <code>
            <span>
              :sparkles:
            </span>
          </code>
        </button>
        <p>
          Introduce new features.
        </p>
      </div>
    </div>
  </article>
</div>
`;

exports[`GitmojiList when is not list mode should render the component 1`] = `
<div
  className="row"
  id="gitmoji-list"
>
  <div
    className="col-xs-12"
  >
    <div
      className="container"
    >
      <div
        className="inputWrapper"
      >
        <input
          className="searchInput"
          name="searchInput"
          onChange={[Function]}
          placeholder="Search your gitmoji..."
          type="search"
          value=""
        />
      </div>
      <div
        className="actionsContainer"
      >
        <div
          className="container"
        >
          <button
            className="button"
            disabled={true}
          />
        </div>
        <div
          className="container"
        >
          <button
            className="button buttonActive"
            disabled={true}
            onClick={[Function]}
          >
            <svg
              className="icon icon-grid"
            >
              <use
                xlinkHref="#icon-grid"
              />
            </svg>
          </button>
          <button
            className="button "
            disabled={false}
            onClick={[Function]}
          >
            <svg
              className="icon icon-list"
            >
              <use
                xlinkHref="#icon-list"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
  <article
    className="emoji col-xs-12 col-sm-6 col-md-3"
    style={
      {
        "--emojiColor": "#ff7281",
      }
    }
  >
    <div
      className="card "
    >
      <header
        className="cardHeader"
      >
        <button
          className="gitmoji-clipboard-emoji gitmoji"
          data-clipboard-text="üé®"
          type="button"
        >
          üé®
        </button>
      </header>
      <div
        className="gitmojiInfo"
      >
        <button
          className="gitmoji-clipboard-code gitmojiCode"
          data-clipboard-text=":art:"
          tabIndex={-1}
          type="button"
        >
          <code>
            <span>
              :art:
            </span>
          </code>
        </button>
        <p>
          Improve structure / format of the code.
        </p>
      </div>
    </div>
  </article>
  <article
    className="emoji col-xs-12 col-sm-6 col-md-3"
    style={
      {
        "--emojiColor": "#40c4ff",
      }
    }
  >
    <div
      className="card "
    >
      <header
        className="cardHeader"
      >
        <button
          className="gitmoji-clipboard-emoji gitmoji"
          data-clipboard-text="‚ö°Ô∏è"
          type="button"
        >
          ‚ö°Ô∏è
        </button>
      </header>
      <div
        className="gitmojiInfo"
      >
        <button
          className="gitmoji-clipboard-code gitmojiCode"
          data-clipboard-text=":zap:"
          tabIndex={-1}
          type="button"
        >
          <code>
            <span>
              :zap:
            </span>
          </code>
        </button>
        <p>
          Improve performance.
        </p>
      </div>
    </div>
  </article>
  <article
    className="emoji col-xs-12 col-sm-6 col-md-3"
    style={
      {
        "--emojiColor": "#ff9d44",
      }
    }
  >
    <div
      className="card "
    >
      <header
        className="cardHeader"
      >
        <button
          className="gitmoji-clipboard-emoji gitmoji"
          data-clipboard-text="üî•"
          type="button"
        >
          üî•
        </button>
      </header>
      <div
        className="gitmojiInfo"
      >
        <button
          className="gitmoji-clipboard-code gitmojiCode"
          data-clipboard-text=":fire:"
          tabIndex={-1}
          type="button"
        >
          <code>
            <span>
              :fire:
            </span>
          </code>
        </button>
        <p>
          Remove code or files.
        </p>
      </div>
    </div>
  </article>
  <article
    className="emoji col-xs-12 col-sm-6 col-md-3"
    style={
      {
        "--emojiColor": "#8cd842",
      }
    }
  >
    <div
      className="card "
    >
      <header
        className="cardHeader"
      >
        <button
          className="gitmoji-clipboard-emoji gitmoji"
          data-clipboard-text="üêõ"
          type="button"
        >
          üêõ
        </button>
      </header>
      <div
        className="gitmojiInfo"
      >
        <button
          className="gitmoji-clipboard-code gitmojiCode"
          data-clipboard-text=":bug:"
          tabIndex={-1}
          type="button"
        >
          <code>
            <span>
              :bug:
            </span>
          </code>
        </button>
        <p>
          Fix a bug.
        </p>
      </div>
    </div>
  </article>
  <article
    className="emoji col-xs-12 col-sm-6 col-md-3"
    style={
      {
        "--emojiColor": "#fb584a",
      }
    }
  >
    <div
      className="card "
    >
      <header
        className="cardHeader"
      >
        <button
          className="gitmoji-clipboard-emoji gitmoji"
          data-clipboard-text="üöëÔ∏è"
          type="button"
        >
          üöëÔ∏è
        </button>
      </header>
      <div
        className="gitmojiInfo"
      >
        <button
          className="gitmoji-clipboard-code gitmojiCode"
          data-clipboard-text=":ambulance:"
          tabIndex={-1}
          type="button"
        >
          <code>
            <span>
              :ambulance:
            </span>
          </code>
        </button>
        <p>
          Critical hotfix.
        </p>
      </div>
    </div>
  </article>
  <article
    className="emoji col-xs-12 col-sm-6 col-md-3"
    style={
      {
        "--emojiColor": "#ffe55f",
      }
    }
  >
    <div
      className="card "
    >
      <header
        className="cardHeader"
      >
        <button
          className="gitmoji-clipboard-emoji gitmoji"
          data-clipboard-text="‚ú®"
          type="button"
        >
          ‚ú®
        </button>
      </header>
      <div
        className="gitmojiInfo"
      >
        <button
          className="gitmoji-clipboard-code gitmojiCode"
          data-clipboard-text=":sparkles:"
          tabIndex={-1}
          type="button"
        >
          <code>
            <span>
              :sparkles:
            </span>
          </code>
        </button>
        <p>
          Introduce new features.
        </p>
      </div>
    </div>
  </article>
</div>
`;
